ui/lib/kv/addon/components/page/secrets/create.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+/**
+ * @module KvSecretCreate is used for creating the initial version of a secret
+ *
+ * <Page::Secrets::Create
+ *    @secret={{this.model.secret}}
+ *    @metadata={{this.model.metadata}}
+ *    @breadcrumbs={{this.breadcrumbs}}
+ *  />
+ *
+ * @param {model} secret - Ember data model: 'kv/data', the new record saved by the form
+ * @param {model} metadata - Ember data model: 'kv/metadata'
+ * @param {array} breadcrumbs - breadcrumb objects to render in page header
+ */
+
+export default class KvSecretCreate extends Component {
+  @service flashMessages;
+  @service router;
+
+  @tracked showJsonView = false;
+  @tracked errorMessage;
+  @tracked modelValidations;
+  @tracked invalidFormAlert;
+
+  @action
+  toggleJsonView() {
+    this.showJsonView = !this.showJsonView;
+  }
+
+  @action
+  pathValidations() {
+    // check path attribute warnings on key up
+    const { state } = this.args.secret.validate();
+    if (state?.path?.warnings) {
+      // only set model validations if warnings exist
+      this.modelValidations = state;
+    }
+  }
+
+  @task
+  *save(event) {
+    event.preventDefault();
+    try {
+      const { isValid, state, invalidFormMessage } = this.args.secret.validate();
+      this.modelValidations = isValid ? null : state;
+      this.invalidFormAlert = invalidFormMessage;
+      if (isValid) {
+        const { secret } = this.args;
+        yield this.args.secret.save();
+        this.flashMessages.success(`Successfully created secret ${secret.path}.`);
+        this.router.transitionTo('vault.cluster.secrets.backend.kv.secret.details', this.args.secret.path);
+      }
+    } catch (error) {
+      const message = error.errors ? error.errors.join('. ') : error.message;
+      this.errorMessage = message;
+      this.invalidFormAlert = 'There was an error submitting this form.';
+    }
+  }
+
+  @action
+  onCancel() {
+    this.router.transitionTo('vault.cluster.secrets.backend.kv.list');
```

Comment:
If they were creating a secret from a directory (ex: `beep/`) and they pressed cancel here, do we expect them to go to the main list route, or back to the directory? If it's the later, I believe we'd have to do some additional logic to check if in a nested directory and navigate back to `list-directory` instead of `list`.

Currently, we don't account for this flow (we direct back to the main list route). Just thought I'd raise the question to see if it's something we think we should account for—maybe as a nice to have, or something design can consider.

Reply:
Yeah - I wondered about this. I think it'd be a nice to have - if the current functionality is to just go back to the list route

Reply:
Resolved in next commit!